### PR TITLE
Upgrade pylint

### PR DESCRIPTION
**Pull request by bmw**
This PR upgrades pylint following the plan [here](https://github.com/certbot/certbot/issues/9441#issuecomment-1302627664) although in addition I changed `certbot/setup.py` so we only declare a pylint dependency for Python >= 3.7.2 to avoid Poetry trying to pin a different version for use with Python < 3.7.2.

I ran the full test suite with these changes at https://dev.azure.com/certbot/certbot/_build/results?buildId=6086&view=results, although it was based on https://github.com/certbot/certbot/pull/9459 before it was squashed and merged.

The ugliest thing to review here is the `.pylintrc` changes. Pylint dropped support for a number of the configuration options we had specified in our ancient pylintrc file. I strongly recommend not reviewing the diff here and instead doing what I did:

1. On `master`, run `git diff a272a0a6e95a36aed01c9ac00701da5a4e37e8d4 .pylintrc` to see the changes we've made to the initial file added in 2014(!).
2. Run `tools/venv.py` to recreate your virtual environment to ensure Pylint 2.15.5 is installed, activate your virtual environment, and then run `pylint --generate-rcfile > .pylintrc`. Your file should now be identical to mine in https://github.com/certbot/certbot/commit/f2519c58c896a998684e7881d082df621f5e278b.
3. Check out this branch and run `git diff f2519c58c896a998684e7881d082df621f5e278b .pylintrc` to see the changes made here to a modern, default pylintrc file and compare this diff to the one you created in step (1).

I hope that's not too annoying to review, but I think the diff created by fixing up the ancient file so that it's compatible with a modern version of pylint would have been almost as bad and this upgrade hopefully results in less work for ourselves in the future.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -13,317 +68,507 @@ jobs=0
...
 
-# Required attributes for module, separated by a comma
-required-attributes=
+# Regular expression matching correct constant names. Overrides const-naming-
+# style. If left empty, constant names will be checked with the set naming
+# style.
+#const-rgx=
 
-# List of builtins function names that should not be used, separated by a comma
-bad-functions=map,filter,apply,input,file
```
Comment (alexzorin) [inline .pylintrc:116]: Why was this removed?

Comment (bmw) [reply .pylintrc:116]: Because at least on my laptop when I add it back to the basics section pylint prints:

> ************* Module .pylintrc
> .pylintrc:1:0: E0015: Unrecognized option found: bad-functions (unrecognized-option)

Comment (bmw) [reply .pylintrc:116]: I think what's going on here is this is a setting for an [optionally enabled extension](https://pylint.pycqa.org/en/latest/user_guide/configuration/all-options.html#deprecated-builtins-checker) and pylint just started complaining because we don't have that checker enabled.

If I add use of one of these functions to our code in `master`, I also have to change the `load-plugins` to:
```
load-plugins=linter_plugin,pylint.extensions.bad_builtin
```
for pylint to actually complain about it.

Comment (alexzorin) [reply .pylintrc:116]: I see what you mean. The rule was never effective to begin with, but [Pylint got stricter about unrecognized options in 2.14](https://github.com/PyCQA/pylint/commit/6efa821fb68bbc8263c3fcacb29db08a8944d60c).

If we never had it, I don't care to add it now.

**Review by alexzorin — APPROVED** [review]
LGTM
